CMakeLists.txt

```diff
@@ -229,30 +225,84 @@ elseif (${HOST_OS} EQUAL ${OS_LINUX} OR ${HOST_OS} EQUAL ${OS_ANDROID})
     ) # todo: configure linux audio lib options
   
     if(NOT ANDROID)
-      list(APPEND osd_SRCS 
-		${d_core}/linux-dist/x11.cpp
-		${d_core}/linux-dist/main.cpp
-		${d_core}/linux-dist/evdev.cpp)
+      list(APPEND osd_SRCS ${d_core}/linux-dist/main.cpp)
 
-      add_definitions(-DSUPPORT_X11)  ## don't use GLES ?
-      link_libraries(X11)
+      link_libraries(${CMAKE_DL_LIBS})
 
-      find_package(PkgConfig)
-      pkg_check_modules(EGL egl)
-      if(EGL_FOUND)
+      find_package(OpenGL)
+
+      if(OpenGL_EGL_FOUND)
         add_definitions(-DSUPPORT_EGL)
+        list(APPEND osd_SRCS ${d_core}/utils/glinit/egl/egl.cpp)
+        link_libraries(OpenGL::EGL)
+      endif()
+
+      find_package(PkgConfig)
+
+      pkg_check_modules(ALSA alsa)
+      if(ALSA_FOUND)
+        add_definitions(-DUSE_ALSA)
+        list(APPEND osd_SRCS ${d_aout}/audiobackend_alsa.cpp)
+        include_directories(${ALSA_INCLUDE_DIRS})
+        link_libraries(${ALSA_LIBRARIES})
+      endif()
+
+      pkg_check_modules(AO ao)
+      if(AO_FOUND)
+        add_definitions(-DUSE_LIBAO)
+        list(APPEND osd_SRCS ${d_aout}/audiobackend_libao.cpp)
+        include_directories(${AO_INCLUDE_DIRS})
+        link_libraries(${AO_LIBRARIES})
+      endif()
+
+      pkg_check_modules(LIBEVDEV libevdev)
+      if(LIBEVDEV_FOUND)
+        add_definitions(-DUSE_EVDEV)
+        list(APPEND osd_SRCS ${d_core}/linux-dist/evdev.cpp)
+        include_directories(${LIBEVDEV_INCLUDE_DIRS})
+        link_libraries(${LIBEVDEV_LIBRARIES})
+      endif()
+
+      pkg_check_modules(LIBPULSE_SIMPLE libpulse-simple)
+      if(LIBPULSE_SIMPLE_FOUND)
+        add_definitions(-DUSE_PULSEAUDIO)
+        list(APPEND osd_SRCS ${d_aout}/audiobackend_pulseaudio.cpp)
+        include_directories(${LIBPULSE_SIMPLE_INCLUDE_DIRS})
+        link_libraries(${LIBPULSE_SIMPLE_LIBRARIES})
+      endif()
+
+      pkg_check_modules(LIBUDEV libudev)
+      if(LIBUDEV_FOUND)
+        add_definitions(-DUSE_UDEV)
+        include_directories(${LIBUDEV_INCLUDE_DIRS})
+        link_libraries(${LIBUDEV_LIBRARIES})
+      endif()
+
+      pkg_check_modules(X11 REQUIRED x11)
+      if(X11_FOUND)
+        add_definitions(-DSUPPORT_X11)  ## don't use GLES ?
+        list(APPEND osd_SRCS ${d_core}/linux-dist/x11.cpp)
+        include_directories(${X11_INCLUDE_DIRS})
+        link_libraries(${X11_LIBRARIES})
+        if(OpenGL_GLX_FOUND)
+          add_definitions(-DSUPPORT_GLX)
+          list(APPEND osd_SRCS ${d_core}/utils/glinit/glx/glx.cpp)
+          link_libraries(OpenGL::GLX)
+        endif()
       endif()
     else()
       list(APPEND osd_SRCS 
         .//android-studio/reicast/src/main/jni/src/Android.cpp
         .//android-studio/reicast/src/main/jni/src/utils.cpp
     #   .//android-studio/reicast/src/main/jni/src/XperiaPlay.c
+        ${d_core}/utils/glinit/egl/egl.cpp
       )
+
+      add_definitions(-DUSE_EVDEV)
+      link_libraries(asound dl EGL Xext)
```

Comment:
Aren't `asound`, `EGL` and `Xext` handled by the auto-detection logic?

Reply:
I have only tested my changes on Linux. I don't know how Android works so I kept this line to not break things on this system.

Reply:
Android uses the standard makefile system, the cmake/android path is not used currently. CMake is used for windows right now.

Reply:
Would it be better to remove the cmake/android path then? I don't mind to keep these lines as a hint for a potential Android support.

Reply:
Yes, it is best to remove it. We can re-add it in a clean way when things are ready for it

Reply:
Right, android path removed